src/@batch-flask/core/data/list-getter/list-getter.ts

```diff
@@ -70,38 +70,39 @@ export abstract class ListGetter<TEntity extends Record<any>, TParams> extends G
             return of(cachedResponse);
         }
 
-        return this.list(params, options).pipe(map(x => this._processItems(cache, x, params, options, true)));
+        return this.list(params, options).pipe(flatMap(x => this._processItems(cache, x, params, options, true)));
```

Comment:
prefer switchMap for this as it will support cancelling. Flatmap will keep making calls until its done. So here if you were to load something and it would take 5min but go away to another page it would keep try loading calls after calls.